firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -426,15 +428,56 @@ private FieldFilter parseFieldFilter(@NonNull FieldPath fieldPath, Operator op,
               .parseQueryValue(value, op == Operator.IN || op == Operator.NOT_IN);
     }
     FieldFilter filter = FieldFilter.create(fieldPath.getInternalPath(), op, fieldValue);
-    validateNewFilter(filter);
     return filter;
   }
 
+  /**
+   * Takes a {@link Filter.CompositeFilter} object, parses each of its subfilters, and a new {@link
```

Comment:
Word missing before "a new". The return type in the comment is also not correct.